lib/src/macro.dart

```diff
@@ -11,37 +11,37 @@ abstract class DeclarationMacro implements Macro {}
 abstract class DefinitionMacro implements Macro {}
 
 abstract class ClassTypeMacro implements TypeMacro {
-  void type(TargetClassType type);
+  void forClassType(TargetClassType type);
 }
 
 abstract class ClassDeclarationMacro implements DeclarationMacro {
-  void declare(TargetClassDeclaration declaration);
+  void forClassDeclaration(TargetClassDeclaration declaration);
```

Comment:
Here's an idea: If a phase two macro on a class declares a method, then in phase 3, the compiler will automatically directly call `forMethodDefinition()` for that method. Likewise for fields.

This way, a macro that both declares and defines methods doesn't have to manually iterate through all the methods in phase 3 to find the ones it itself declared. What do you think?

Or maybe in the API for declaring a method, the macro can pass in a callback that gets invoked in phase 3 to define that method?

Reply:
A callback might work, you could also just add the macro application of yourself to the code as well which in effect is the same thing - I know you haven't been a huge fan of that but to me that is pretty intuitive, and its one less api we have to define and maintain which I think is a non-trivial bonus 🤷‍♂️ .